### PR TITLE
Visualize distances with color; improve resize behavior

### PR DESCRIPTION
**Pull request by ebellbog**
In standard mode (i.e. not screensaver), click any cell to create a colorful, high-contrast visualization of its distance from all other cells. Click background to reset.

Distances are calculated using a quick-and-dirty breadth-first search, implemented on the `Maze` class.

### CODE REVIEW

**Comment by ebellbog** [comment]
Screen Shot 2019-03-16 at 1 05 56 AM

File: src/js/Maze.js [hunk 1]
```diff
@@ -51,6 +51,42 @@ class Maze {
...
+                    unvisited.distance = cell.distance + 1;
+                    maxDist = Math.max(maxDist, unvisited.distance);
+                    newFrontier.push(unvisited);
+                });
+            });
+            frontierCells = newFrontier;
+        }
+
+        // Set hue by normalized distance (i.e. fraction of max dist).
+        cells.forEach((cell) => {
```
Comment (rye-welz-geselowitz) [inline src/js/Maze.js]: I'm sure you have a reason for this, but it feels a little counter-intuitive that close=red. Usually I think of red as marking danger/inaccessiblity/difficulty, so I feel like maybe the hardest to reach cells should be red? 

I also wonder if doing it in gray scale would be more visually effective. For me at least, the colors are cognitively a little hard to follow

File: src/js/Maze.js [hunk 2]
```diff
@@ -51,6 +51,42 @@ class Maze {
...
+
+    /**
+     * Calculate distance of all maze cells from cell at given index,
+     * then use color to visualize distance.
+     *
+     * @param {number} cellIdx - Index of selected start cell.
+     */
+    colorByDistanceFromCell(cellIdx) {
+        const cells = this.getCells();
+        cells.forEach((cell) => cell.distance = null);
```
Comment (rye-welz-geselowitz) [inline src/js/Maze.js]: I still think I have a slight preference for tracking this in a dictionary as opposed to on the cell object - I don't really like putting extra properties on something that are incidental to actual existence. Then you could also skip initializing them as null and just have 
```
const distancesFromCell = {}
```

File: src/js/Maze.js [hunk 3]
```diff
@@ -51,6 +51,42 @@ class Maze {
...
+
+        const startCell = cells[cellIdx];
+        startCell.distance = 0;
+
+        let maxDist = 0;
+        let frontierCells = [startCell];
+
+        while (frontierCells.length) {
```
Comment (rye-welz-geselowitz) [inline src/js/Maze.js:68]: yay you did BFS with a while loop!!!! Weren't we talking about how you usually do it with recursion haha?

File: src/js/Maze.js [hunk 4]
```diff
@@ -51,6 +51,42 @@ class Maze {
...
+            const newFrontier = [];
+            frontierCells.forEach((cell) => {
+                const accessibleNeighbors = cell.getAccessibleNeighbors();
+                const unvisitedAccessible = accessibleNeighbors.filter((neighbor) => neighbor.distance === null);
```
Comment (rye-welz-geselowitz) [inline src/js/Maze.js]: The fact that we have methods with the word "visited", which means a totally different thing, on our cell, makes me wonder if we need to change those method names. I feel like the constant re-use of the word "visited" with different meanings would be super confusing for someone jumping into the codebase. 🤔 Definitely seems appropriate here though!!

File: src/js/Maze.js [hunk 5]
```diff
@@ -51,6 +51,42 @@ class Maze {
...
         }
     }
+
+    /**
+     * Calculate distance of all maze cells from cell at given index,
+     * then use color to visualize distance.
+     *
+     * @param {number} cellIdx - Index of selected start cell.
+     */
+    colorByDistanceFromCell(cellIdx) {
```
Comment (rye-welz-geselowitz) [inline src/js/Maze.js]: I know another purpose of this function was to help us find the shortest path. I also think functions on the maze should be responsible for processing/reporting INFORMATION about the maze, not for rendering it. Therefore I wonder if it would make sense (not now necessarily, but soon) to
(1) Change its name to something like "getDistancesFromCellDictionary"
(2) Have it return a dictionary mapping cells (or cell coordinates) to the distance from the cell
(3) Then, call that function and do the coloring logic in `index.js`, since I don't think this kind of styling action really belongs on the maze object anyway

**Review by rye-welz-geselowitz — APPROVED** [review]
Ahhhhh it looks so good, I love it!! And your implementation is really simple and straightforward, I love that something so visually impactful really does take only a few lines of code.
